apps/learning_nuggets/models.py

```diff
@@ -0,0 +1,126 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.urls import reverse
+from django.utils.text import slugify
+from wagtail.admin.panels import FieldPanel
+from wagtail.fields import StreamField
+from wagtail.models import Page
+from wagtail.search import index
+from wagtail.snippets.models import register_snippet
+
+from .blocks import LearningContentBlock
+
+
+@register_snippet
+class LearningCategory(models.Model):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField(unique=True, blank=True)
+    permission_level = models.CharField(
+        max_length=50,
+        choices=[
+            ("participant", "Participant"),
+            ("initiator", "Initiator"),
+            ("moderator", "Moderator"),
+        ],
+        default="participant",
+    )
+    order = models.IntegerField(default=0)
+
+    panels = [
+        FieldPanel("name"),
+        FieldPanel("slug"),
+        FieldPanel("permission_level"),
+        FieldPanel("order"),
+    ]
+
+    class Meta:
+        ordering = ["order", "name"]
+        verbose_name = "Learning Category"
+        verbose_name_plural = "Learning Categories"
+
+    def save(self, update_fields=None, *args, **kwargs):
+        if not self.slug:
+            base_slug = slugify(self.name)
+            unique_slug = base_slug
+            counter = 1
+
+            while LearningCategory.objects.filter(slug=unique_slug).exists():
+                unique_slug = f"{base_slug}-{counter}"
+                counter += 1
+
+            self.slug = unique_slug
+
+        if update_fields:
+            update_fields = {"slug"}.union(update_fields)
+        super().save(update_fields=update_fields, *args, **kwargs)
```

Comment:
in django 4,2 and beyond we need to add the `updated_fields` parameter when we customize the save() method. It looks for this field when it updates an object,